File: src/test/java/pl/patrykdolata/lifemanager/security/AuthenticationUtilsTest.java
package pl.patrykdolata.lifemanager.security;

import org.junit.jupiter.api.Test;
import org.springframework.security.test.context.support.WithAnonymousUser;
import pl.patrykdolata.lifemanager.TestUtils;
import pl.patrykdolata.lifemanager.domain.UserEntity;
import pl.patrykdolata.lifemanager.exceptions.NotLoggedUserException;

import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.is;
import static org.junit.jupiter.api.Assertions.assertNotNull;
import static org.junit.jupiter.api.Assertions.assertThrows;

class AuthenticationUtilsTest {

    @WithAnonymousUser
    @Test
    void getCurrentUserId_whenNullAuthentication_expectException() {
        assertThrows(NotLoggedUserException.class, AuthenticationUtils::getCurrentUserId);
    }

    @Test
    void getCurrentUserId_whenStringPrincipal_expectException() {
        TestUtils.withStringPrincipal();
        assertThrows(NotLoggedUserException.class, AuthenticationUtils::getCurrentUserId);
    }

    @Test
    void getCurrentUserId_whenProperPrincipal_expectId() {
        TestUtils.withProperPrincipal();
        Long result = AuthenticationUtils.getCurrentUserId();
        assertThat(result, is(TestUtils.ID));
    }

    @Test
    void getCurrentUser_whenProperPrincipal_expectUser() {
        TestUtils.withProperPrincipal();
        UserEntity result = AuthenticationUtils.getCurrentUser();
        assertNotNull(result);
        assertThat(result.getId(), is(TestUtils.ID));
    }
}

File: src/test/kotlin/pl/patrykdolata/lifemanager/security/AuthenticationUtilsTest.kt
package pl.patrykdolata.lifemanager.security

import org.hamcrest.MatcherAssert.assertThat
import org.hamcrest.Matchers.equalTo
import org.junit.jupiter.api.Assertions.assertNotNull
import org.junit.jupiter.api.Assertions.assertThrows
import org.junit.jupiter.api.Test
import pl.patrykdolata.lifemanager.TestUtils
import pl.patrykdolata.lifemanager.exceptions.NotLoggedUserException
import pl.patrykdolata.lifemanager.security.AuthenticationUtils.getCurrentUser
import pl.patrykdolata.lifemanager.security.AuthenticationUtils.getCurrentUserId

class AuthenticationUtilsTest {

    @Test
    fun `Given null authentication should throw exception`() {
        TestUtils.withNullAuthentication()
        assertThrows(NotLoggedUserException::class.java) { getCurrentUserId() }
    }

    @Test
    fun `Given string principal should throw exception`() {
        TestUtils.withStringPrincipal()
        assertThrows(NotLoggedUserException::class.java) { getCurrentUserId() }
    }

    @Test
    fun `Given proper principal should return proper user id`() {
        TestUtils.withProperPrincipal()
        val result = getCurrentUserId()
        assertThat(result, equalTo(TestUtils.ID))
    }

    @Test
    fun `Given proper principal should return proper user entity`() {
        TestUtils.withProperPrincipal()
        val result = getCurrentUser()
        assertNotNull(result)
        assertThat(result.id, equalTo(TestUtils.ID))
    }
}
